File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3079_find-the-sum-of-encrypted-integers/solutions/java/solution.java
class Solution {
    public int sumOfEncryptedInt(int[] nums) {
        int sum = 0;
        for (int num : nums) {
            int maxDigit = 0;
            int temp = num;
            while (temp > 0) {
                maxDigit = Math.max(maxDigit, temp % 10);
                temp /= 10;
            }
            int encryptedNum = 0;
            int place = 1;
            while (num > 0) {
                encryptedNum += maxDigit * place;
                num /= 10;
                place *= 10;
            }
            sum += encryptedNum;
        }
        return sum;
    }
}

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3079_find-the-sum-of-encrypted-integers/solutions/python3/solution.py
class Solution:
    def sumOfEncryptedInt(self, nums: list[int]) -> int:
        def encrypt(x):
            str_x = str(x)
            max_digit = max(str_x)
            return int(max_digit * len(str_x))
        
        return sum(encrypt(num) for num in nums)
